From df6c160bdde145a275d9e94691a0262f9ea9a09c Mon Sep 17 00:00:00 2001
From: Johan Hovold <johan+linaro@kernel.org>
Date: Thu, 29 Jun 2023 09:58:19 +0200
Subject: [PATCH 010/173] hack: arm64: disable pointer authentication

Disable pointer authentication which prevents sc8280xp from booting.

This is needed until the firmware has been fixed. An alternative is to
add 'arm64.nopauth' to the kernel command line.

Not-signed-off-by: Johan Hovold <johan+linaro@kernel.org>
---
 arch/arm64/kernel/idreg-override.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/kernel/idreg-override.c b/arch/arm64/kernel/idreg-override.c
index 3addc09f8746..163161fea17a 100644
--- a/arch/arm64/kernel/idreg-override.c
+++ b/arch/arm64/kernel/idreg-override.c
@@ -273,9 +273,16 @@ static __init void __parse_cmdline(const char *cmdline, bool parse_aliases)
 
 		match_options(buf);
 
-		for (i = 0; parse_aliases && i < ARRAY_SIZE(aliases); i++)
-			if (parameq(buf, aliases[i].alias))
+		for (i = 0; parse_aliases && i < ARRAY_SIZE(aliases); i++) {
+			/*
+			 * HACK: Disable pointer authentication until the
+			 * sc8280xp firmware has been fixed.
+			 */
+			if (parameq(buf, aliases[i].alias) ||
+			    strcmp(aliases[i].alias, "arm64.nopauth") == 0) {
 				__parse_cmdline(aliases[i].feature, false);
+			}
+		}
 	} while (1);
 }
 
-- 
2.43.0

